axum-core/src/macros.rs

```diff
@@ -132,7 +132,7 @@ macro_rules! __define_rejection {
 
         impl std::fmt::Display for $name {
             fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-                write!(f, "{}", $body)
+                write!(f, concat!($body, ": {}"), self.0)
```

Comment:
I'd slightly prefer `f.write_str` to not rely on inlining as much, but that probably affects other places too and can be a separate PR once again.

Reply:
I've added another commit that switches it over to use `write_str()`

Reply:
Ah, I thought it was just going to be one `write_str`, but I guess that would have only been the case for the non-composite case. Anyways, doesn't hurt.